language/move-lang/src/errors/new.rs

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::errors::diagnostic_codes::{DiagnosticCode, DiagnosticInfo};
+use codespan_reporting_new::{
+    self as csr,
+    files::SimpleFiles,
+    term::{emit, termcolor::WriteColor, Config},
+};
+use move_ir_types::location::*;
+use std::{
+    collections::{HashMap, HashSet},
+    iter::FromIterator,
+    ops::Range,
+};
+
+//**************************************************************************************************
+// Types
+//**************************************************************************************************
+
+pub type FileId = usize;
+
+pub type FilesSourceText = HashMap<&'static str, String>;
+type FileMapping = HashMap<&'static str, FileId>;
+
+#[derive(PartialEq, Eq, Clone, Copy, Debug, Hash)]
+pub enum Severity {
+    Error,
+    Warning,
+}
```

Comment:
Nit: this might make more sense being defined in `diagnostic_codes`

Reply:
That actually isn't used anymore. Will remove